Combat harmonization (#192)
creatis-myriad/creatis-myriad.github.io: collections/_posts/2024-11-28-ComBat-Harmonization.md

---
 layout: review
 title: ComBat, Harmonization of multi-site diffusion tensor imaging data
 tags: statistics ComBat harmonization diffusion medical DTI multi-site inter-scanner
 cite:
     authors: "Jean-Philippe Fortin and Drew Parker and Birkan Tunç and Takanori Watanabe and Mark A. Elliott and Kosha Ruparel and David R. Roalf and Theodore D. Satterthwaite and Ruben C. Gur and Raquel E. Gur and Robert T. Schultz and Ragini Verma and Russell T. Shinohara"
     title:   "Harmonization of multi-site diffusion tensor imaging data"
     venue:   "article from NeuroImage"
 pdf: "https://www.biorxiv.org/content/10.1101/116541v3.full.pdf"
 ---

# Highlights

* Significant site and scanner effects exist in DTI scalar maps.
* Several multi-site harmonization methods are proposed.

> * ComBat performs the best at removing site effects in FA and MD.

Reviewer: same comment for the other metrics

## Harmonisation methods

The authors use and adapt five statistical harmonization techniques for DTI data: **global scaling**, **functional normalization**[^2], **RAVEL**[^3], **Surrogate Variable Analysis** (SVA)[^4][^5] and **ComBat**[^6].
*Raw data* means that no harmonization was performed.

The RAVEL algorithm attempts to estimate a voxel-specific unwanted variation term by using a control region of the brain to estimate latent factors of unwanted variation common to all voxels.

The SVA algorithm estimates latent factors of unwanted variation, called surrogate variables, that are not associated with the biological covariates of interest. It is particularly useful when the site variable is not known, or when there exists residual unwanted variation after the removal of site effects

### ComBat
ComBat model was introduced in the context of gene expression analysis by Johnson et al. [2007] [^6] as an improvement of location/scale models for studies with small sample size.
The ComBat model is reformulated in the context of DTI images.
Let $m$ the number of imaging sites, containing $n_i$ for $i=1,2,...,m$ volume scans.

> For voxel $\nu=1,2,...,p$, let $y_{ij\nu}$ represent the diffusion metric measure (FA for e.g.) at voxel $\nu$ for the scan $j$ at site $i$. ComBat posits the fllowing location and scale (L/S) adjustement model:

Reviewer: ```suggestion
For voxel $\nu=1,2,...,p$, let $y_{ij\nu}$ represent the diffusion metric measure (FA for e.g.) at voxel $\nu$ for the scan $j$ at site $i$. ComBat posits the following location and scale (L/S) adjustement model:
```

More specifically, the site effects parameters are assumed to have the parametric prior distributions:

$$ \gamma_{i\nu}\sim \mathcal{N}(\gamma_i, \tau_i^2) \qquad and \qquad \delta_{i\nu}^2 \sim InverseGamma(\lambda_i,\theta_i) \qquad (3) $$

The hyperparameters $\gamma_i$, $\tau_i^2$, $\lambda_i$, $\theta_i$ are estimated empirically from the data as described in [^6]. The ComBat estimates $\gamma_{i\nu}^*$ and $\delta_{i\nu}^*$ of the site effect parameters are computed using conditional posterior means.

The final ComBat-harmonized diffusion metric values are defined as:

$$ y_{ijv}^{ComBat} = \frac{y_{ij\nu}-\hat\alpha_\nu -X_{ij}\hat\beta_\nu - \gamma_{i\nu}^*}{\delta_{i\nu}^*} + \hat\alpha_\nu + X_{ij}\hat\beta_\nu \qquad (4)$$

### How does it work in practice ?
Another bibliography session ?

> <!-- Show how the compute the estimates with the bayesian framework -->

Reviewer: ```suggestion
<!-- Show how to compute the estimates with the bayesian framework -->
```

*Replicability is defined as the chance that an independent experiment will produce a similar set of results*, and is a strong indication that a set of results is biologically meaningful.
Associations with age are measured using usual Wald t-statistics from linear regression. They test the replicability of the voxels associated with age using a discovery-validation scheme.
<!-- to be explained, maybe more details from the paper -->

## Creation of silver-standards

To further evaluate the performance of the different harmonization methods, two sets of silver-standards per site were created:
1) A silver-standard for voxels that are truly associated with age (signal silver-standard).
2) A silver-standard for voxels that are not associated with age (null silver-standard).

For each silvert-standard two different sets are created since previous studies have shown that some brain regions are more specific in changes for FA and others for MD.

To estimate them, the authors use a meta-analytic approach: 
1) for each site separatly, at each voxel in the WM, they apply a linear regression model to obtain a t-statistic measuring the association of FA with age.

> 2) For each site, they define the site-specific signal silver-standard to be the k=5000 voxels with the highest t-sttistics in magnitude.

Reviewer: ```suggestion
2) For each site, they define the site-specific signal silver-standard to be the k=5000 voxels with the highest t-statistics in magnitude.
```

This process ensures that the resulting voxels are not only voxels highly associated with age within a study, but also replicated across the two sites. Voxels obtained are consistent with the litterature.

Similarly the authors estimate the silver standard for voxels not associated with by age when considering the k=5000 voxels with the lowest t-statistics by site.

# Results

## DTI scalar maps are higly affected by site
See figure 2

## ComBat successfully removes site effects in DTI scalar maps

<div style="text-align:center">
<img src="/collections/images/comBat/fig3.jpg" width=800></div>

Figure 3 shows the MA-plots before and after each harmonization for the FA maps.
* Scaling and Funnorm methods centered the MA-plots around 0, which is consistent with the global nature of their harmonization methods.
* RAVEL, SVA and ComBat reduce greatly the inter-sites differences.
* Ravel does not seem to account for local-site effects.

<!-- understand why Ravel downperforms -->

<!--  -->
<div style="text-align:center">
<img src="/collections/images/comBat/fig4.jpg" width=800></div>

> Figure 4a shows a t-statistic at each voxel to measure the association of the DTI scalar values with site. A voxel is significant is the p-value calculated from the two-sample t-test is less than 0.05, after correcting for multiple comparisons using Bonferroni correction. They authors don't explain how they realize the correction in practice. For more details about how they perform a two-sample t-statistics with DTI data, see Figure A.3.

Reviewer: ```suggestion
Figure 4a shows a t-statistic at each voxel to measure the association of the DTI scalar values with site. A voxel is significant if the p-value calculated from the two-sample t-test is less than 0.05, after correcting for multiple comparisons using Bonferroni correction. They authors don't explain how they realize the correction in practice. For more details about how they perform a two-sample t-statistics with DTI data, see Figure A.3.
```

Additionally, as we can see in Figure 4b, all harmonization methods increase the number of significant voxels associated with age in comparison to the raw data (ComBat presents the highest gain for both FA and MD maps).

## Harmonization and confounding
This part aims to investigate the robustness of the different harmonization techniques to datasets for which age is confounded by site.

* The two main datasets are carefully **matched** for age, gender, ethnicity to minimize potential confounding of those kind of variables with site. Howether matching has limitations as it strongly depends on the overlap of covariates between datasets. Poor overlap means many scans to be excluded, making matching infeasible in many applications where harmonization is needed.

* Howether doing nothing will let an undesirable situation happened where site will be a confounder for the relationship between the DTI values and the phenotypes of interest.
  
* Confounding between age and site presents an additional challenge for harmonization, since removing variation associated site can lead to removing variation associated with age if not done carefully.

> To evaluate the robustness of the different harmonization methods in the presences of statistical  confounding between imaging site and age (that is when age is unbalanced with respect to site), the authors selected different subsets of the data to create confounding scenarios.

Reviewer: ```suggestion
To evaluate the robustness of the different harmonization methods in the presences of statistical confounding between imaging site and age (that is when age is unbalanced with respect to site), the authors selected different subsets of the data to create confounding scenarios.
```

The "effect size" can be estimated using linear regression (slope coefficient associated with age) for each voxel.
To assess unbiasedness without knowing the true effect sizes, the authors curcumvent by estimating the effect sizes on the signal silver-standard (in practice, they compute for each voxel the mean of the slopes they obtain for each site in the silver-standard).

<div style="text-align:center">
<img src="/collections/images/comBat/figB10.jpg" width=500></div>

<div style="text-align:center">
Figure B.10a shows the boxplots of the "effect sizes" from their silvert-standard. In these cohorts, a difference is made between signal voxels and null voxels.
</div>

<div style="text-align:center">
<img src="/collections/images/comBat/fig7.jpg" width=800></div>

<div style="text-align:center">
Figure 7 presents the distribution of the estimated effect sizes on the signal silver-standard for all methods and for all confounding scenarios. Th dashed line represents the median of the true effect sizes, and the solid line represents an effect size of 0.
</div>

> Results for raw data are consitent: positive (resp. negative) confoundind shifts the effect size positively (resp negatively).

Reviewer: ```suggestion
Results for raw data are consitent: positive (resp. negative) confounding shifts the effect size positively (resp negatively).
```

* All methods improve the replicability of the voxels associated with age.
* ComBat performs as well as the silver-standard, successfully removing most of the site effects.

Figure 9.b shows the densities of the t-statistics for the top voxels associated with age (signal voxels) for the FA maps.
* All methods improve the magnitude of the t-statistics, therefore increasing statistical power, with ComBat showing the best performance as well as the silver-standards.

Figure 9.c shows the densities of the t-statistics for the voxels not associated with age (null voxels) for the FA maps.
* A good harmonization method should result in t-statistics centered around 0.
* Performance is variable across harmonization methods. But ComBat is one of those who do perform correctly.

The authors investigated the stability of the ComBat harmonization parameters by running ComBat on random subsamples of variable size. They obtained that site effects estimated from subsamples approximate well the site effects estimated from the full dataset.

# Discussion

> * The authors shows that combining the two studies without proper harmonization led to a decrease in power od detecting voxels associated with age.

Reviewer: ```suggestion
* The authors show that combining the two studies without proper harmonization led to a decrease in power of detecting voxels associated with age.
```

> * The authors shows that combining the two studies without proper harmonization led to a decrease in power od detecting voxels associated with age.

Reviewer 2: power **of** detecting
